project_task_code/models/project_task.py

```diff
@@ -28,3 +29,46 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        args = args or []
+        code = self._name_search_parse_code(name)
+        if not code:
+            return super(ProjectTask, self).name_search(
+                name=name,
+                args=args,
+                operator=operator,
+                limit=limit)
+        # We are searching for codes
+        tasks = self.search(
+            [('code', '=ilike', '%' + code + '%')] + args,
+            limit=limit)
+        if limit and len(tasks) == limit:
+            return tasks.name_get()
+        return tasks.name_get() + super(ProjectTask, self).name_search(
+            name=name,
+            args=args,
+            operator=operator,
+            limit=limit + (limit - len(tasks)) or limit)
+
+    @api.model
+    def _name_search_parse_code(self, name):
+        """ Check if user input looks like code """
+        # Remove any whitespace from name
+        name = name.lstrip()
+        # if user is typing only digits, return right away
+        if name.isdigit():
+            return name
+        # Get the data record
+        sequence_id = self.env.ref('project_task_code.sequence_task')
+        prefix = sequence_id.prefix or ''
+        suffix = sequence_id.suffix or ''
+        # Check if input could be code
+        if not all([
```

Comment:
wait this seems useless. you check that user wrote prefix , number in the middle and suffix before allowing to return name?
prefix + number should already be enough to start to return results..... right?
imagine a user writing a search 'T123', we should already return name....

Reply:
right. We shouldn't care about the suffix at all, as it will never add new information to search, so people also won't type it. If name without prefix is a number, it's a code, that's it.

Thanks for this one!